streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -68,72 +69,115 @@ private static synchronized void updateNumRecordsProduces(final int delta) {
     }
 
     static void generate(final String kafka) {
+        try {
+            Exit.addShutdownHook("streams-eos-test-driver-shutdown-hook", () -> {
+                System.out.println("Terminating");
+                isRunning = false;
+
+                final long timeout = System.currentTimeMillis() + Duration.ofMinutes(5).toMillis();
```

Comment:
Can we use Util.waitUntilCondition here, or even simply using a latch and wait on it instead?